subsys/bluetooth/controller/Kconfig

```diff
@@ -403,21 +403,6 @@ config BT_CTLR_RL_SIZE
 	  Set the size of the Resolving List for LE Controller-based Privacy.
 	  On nRF5x-based controllers, the hardware imposes a limit of 8 devices.
 
-config BT_CTLR_SW_DEFERRED_PRIVACY
-	bool "LE Controller-based Software Privacy"
-	depends on BT_CTLR_PRIVACY
```

Comment:
Just append dependency on BT_LL_SW_SPLIT here, do not move it out of context (grouped along with BT_CTLR_PRIVACY).

Reply:
I do not agree on that we should keep it here. All configurations in this group are generic BLE configurations. All other implementation details are covered in the BT_LL_SW group below. This includes how to handle scheduling, overlapping control procedures, encryptions etc. In my opinion SW-deferred privacy is just another option like this.

Reply:
> All configurations in this group are generic BLE configurations

The Kconfig are always grouped by features, take examples of configurations of host or any other driver features, the Kconfig options stay together. This makes it readable in menuconfig, and to relate the parameters.

> Are we considering BT_LL_SW_SPLIT as a synonym for not out of tree controllers?
No. BT_LL_SW_SPLIT  or BT_LL_SW_LEGACY or any out-of-tree controller are selection options.

@rugeGerritsen You can add new BT_CTLR_SW_DEFERRED_PRIVACY_SUPPORT Kconfig that can be selected by Bluetooth Link Layer Selection.

Reply:
@cvinayak . All configs marked with "SUPPORT" are features that can be used to build the host in a more optimal way. Do you believe BT_CTLR_SW_DEFERRED_PRIVACY_SUPPORT is such an option? I believe that BT_CTLR_PRIVACY_SUPPORT should is sufficient.

Reply:
BT_CTLR_SW_DEFERRED_PRIVACY_SUPPORT removes h/w limit on number of identities to be resolved, hence  yes, it is used to build the host in a more optimal way to support larger number of identities.

@asmk-ot correct me, the BT_CTLR_SW_DEFERRED_PRIVACY_SUPPORT will allow host to add upto 64 IRKs to be resolved by LL?

Reply:
@cvinayak , @asmk-ot , is there a use case to have more than lets say 8 identities?

Reply:
That really depends on how many devices you see in your environment, locking the actual number will certainly represent a trade-off

Reply:
Support for greater than 8 simultaneous connection establishment with limited to 8 IRKs in hardware needs complex application design.

We are deviating from the title of the PR.

NRF has not implemented the LLL required for the SW deferred privacy, if @asmk-ot is ok, then these two Kconfig option can be moved under ADVANCED_FEATURES after the BT_CTLR_FILTER Kconfig option.

Reply:
I'm OK with that, seems sensible for now (although I would hope to see that SW deferred privacy will be implemented on a plethora of platforms in the future)